src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Scopes.cs

```diff
@@ -191,10 +190,16 @@ private class ScopeTracker : IDisposable {
                 // them better.
                 // TODO: figure out threading/locking for the Open/Close pairs.
                 // Debug.Assert(_eval._openScopes.Count > 0, "Attempt to close global scope");
-                if (_eval._openScopes.Count > 0) {
-                    _eval._openScopes.Pop();
+                try {
+                    if (_eval._openScopes.Count > 0) {
+                        _eval._openScopes.Pop();
+                    }
+                    _eval.CurrentScope = _eval._openScopes.Count == 0 ? _eval.GlobalScope : _eval._openScopes.Peek();
+                } catch (InvalidOperationException) {
+                    // Per comment above this can happen occasionally.
+                    // The catch is tactical fix to prevent crashes since complete handling of open/close
+                    // in threaded cases would be much larger change.
```

Comment:
How bad would it be to log the exception (as verbose or trace or something) when this occurs? Just in case this patch hides something and we ask someone to provide a deeper log level (and we find out that it's scope-related stuff).

Reply:
Hate to make you change it again, but can you log the entire exception just so a stack trace gets printed?

Reply:
I thought about stack, but it won't give much info since stack was modified elsewhere. Stack here will always be `HoverSource -> ScopeTracker -> Dispose`

Reply:
Okay.

Reply:
The issue most probably that analysis is running accessing same eval and changes scope while hover is in progress